sdk/src/main/java/com/mapbox/maps/viewannotation/ViewAnnotationManagerImpl.kt

```diff
@@ -213,41 +206,32 @@ internal class ViewAnnotationManagerImpl(
         }
       }
     }
-    // reposition existing views modifying layout parameters
-    addViews(idsToRepositionSet, needsRepositionOnly = true)
-    // add views on screen that were not present before
-    addViews(idsToAddSet, needsRepositionOnly = false)
-    currentViewsDrawnMap.clear()
-    positionDescriptorList.forEach {
-      currentViewsDrawnMap[it.identifier] = it.leftTopCoordinate
-    }
-  }
-
-  private fun addViews(idsSet: HashSet<String>, needsRepositionOnly: Boolean) {
-    idsSet.forEach { id ->
-      annotationMap[id]?.let { annotation ->
-        positionDescriptorMap[id]?.let { viewPosition ->
-          annotation.viewLayoutParams.width = viewPosition.width
-          annotation.viewLayoutParams.height = viewPosition.height
-          annotation.viewLayoutParams.setMargins(
-            viewPosition.leftTopCoordinate.x.toInt(),
-            viewPosition.leftTopCoordinate.y.toInt(),
-            0,
-            0
-          )
-          if (needsRepositionOnly) {
-            // TODO recalculate actual view translation instead or re-requesting layout
-            annotation.view.requestLayout()
-          } else {
-            // remove view to as it may have been not removed before due to visibility
-            mapView.removeView(annotation.view)
-            // removing shadowing effect brought in by setting z-index / elevation
-            annotation.view.outlineProvider = null
-            mapView.addView(annotation.view, annotation.viewLayoutParams)
-          }
-          annotation.view.translationZ = viewPosition.zIndex
+    positionDescriptorList.forEach { viewPosition ->
+      annotationMap[viewPosition.id]?.let { annotation ->
+        annotation.viewLayoutParams.width = viewPosition.width
+        annotation.viewLayoutParams.height = viewPosition.height
+        annotation.viewLayoutParams.setMargins(
+          viewPosition.leftTopCoordinate.x.toInt(),
+          viewPosition.leftTopCoordinate.y.toInt(),
+          0,
+          0
+        )
+        if (viewPosition.needAddToLayout) {
+          // remove view to as it may have been not removed before due to visibility
+          mapView.removeView(annotation.view)
+          mapView.addView(annotation.view, annotation.viewLayoutParams)
```

Comment:
can we just update layout params, but not remove and add?

Reply:
added if clause to make it clearer what's happening.